### PR TITLE
Expose gzip.Reader.Multistream

### PR DESCRIPTION
**Pull request by rgmz**
Somewhat related to #405, it turns out that `compress/gzip` has a feature to handle trailing data without returning an error.

> If you would like the ability to read a valid gzip member and specially handle non-gzip data after the member, then you can use the [gzip.Reader.Multistream](https://pkg.go.dev/compress/gzip#Reader.Multistream) feature to read the file one member at a time.
> 
> https://github.com/golang/go/issues/47809

I created a field called `DisableMultistream` and not `Multistream` because `gzip.Reader.Multistream` is enabled by default. The latter would silently disable `Multistream` for any existing instances of `archiver.Gz{}`. 

**Additional Context**

https://github.com/trufflesecurity/trufflehog/issues/2934

### CODE REVIEW

File: gz.go [hunk 1]
```diff
@@ -65,14 +69,21 @@ func (gz Gz) OpenWriter(w io.Writer) (io.WriteCloser, error) {
...
+		}
+
+		return gzR, err
 	} else {
-		rc, err = gzip.NewReader(r)
+		gzR, err := gzip.NewReader(r)
+		if gzR != nil && gz.DisableMultistream {
+			gzR.Multistream(false)
+		}
+
```
Comment (mholt) [inline gz.go:84]:
```suggestion
```

File: gz.go [hunk 2]
```diff
@@ -65,14 +69,21 @@ func (gz Gz) OpenWriter(w io.Writer) (io.WriteCloser, error) {
...
 func (gz Gz) OpenReader(r io.Reader) (io.ReadCloser, error) {
-	var rc io.ReadCloser
-	var err error
 	if gz.Multithreaded {
-		rc, err = pgzip.NewReader(r)
+		gzR, err := pgzip.NewReader(r)
+		if gzR != nil && gz.DisableMultistream {
+			gzR.Multistream(false)
+		}
+
```
Comment (mholt) [inline gz.go:77]:
```suggestion
```

File: gz.go [hunk 3]
```diff
@@ -65,14 +69,21 @@ func (gz Gz) OpenWriter(w io.Writer) (io.WriteCloser, error) {
...
+		return gzR, err
 	} else {
```
Comment (mholt) [inline gz.go:79]: Since the prior block ends in `return` we can drop the `else` block entirely.

**Review by mholt — CHANGES_REQUESTED** [review]
Thanks! I have the stupidest style nits to pick with this :sweat_smile:  but then we can get it merged in right away.

**Comment by rgmz** [comment]
Done!

**Review by mholt — APPROVED** [review]
Cool -- let's try it. Thanks!

**Pull request merged**